### ISSUE TITLE
error in multiplying/dividing a matrix by a scalar in DSL

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#284]
We have an incompatibility between R and C execution of matrix/scalar multiplication (same issue for division).

The following example works in compiled code but not in R, because model[[scalNode]] is a 1-dim array. 

One kludgey possibility is to wrap a drop() around model[[scalNode]] in such situations. Adding a [1] to get model[[scalNode]][1] messes up the compiled code because model[[scalNode]] is a scalar.

I'll note that this came up in testing contexts before but (I don't think) we realized it would affect DSL calculations when run in R.

I'm happy to work on this but wanted Perry to weigh in on thoughts on what to do about it.

```
mynf <- nimbleFunction(
    setup = function(model, matNode, scalNode) {},
    run = function(mat = double(2)) {
#        tmp = model[[matNode]] * model[[scalNode]]  # same issue if matrix is from a node instead
        tmp = mat * model[[scalNode]]
        returnType(double(2))
        return(tmp)
    })

code <- nimbleCode({
    sigma ~ dunif(0, 5)
    mat2[1:2,1:2] ~ dwish(R[1:2,1:2], 4)
})
m <- nimbleModel(code, inits = list(sigma = 1, R = diag(2)))

rnf <- mynf(m, 'mat2', 'sigma')
rnf$run(diag(2))
#Error in mat * model[[scalNode]] : non-numeric argument to binary operator
cnf = compileNimble(rnf, m, showCompilerOutput = T)
cnf$rnf$run(diag(2))
```

**Comment by perrydv** [comment]
I just published a branch ExperimentalFixToIssue284 that has a fix for this.

Lines 1188-1232 of BUGS_model.R are where the code is generated for active bindings for R model classes.  At some point long ago we had problems with R model variables being set to subtly different types at different times, with non-enforcement during assignment.  So apparently we made everything an array, even a 1-dimensional array of length one rather than a  vector of length one.  Apparently the problem of the current Issue was not noticed.  To fix it, I added a third case of the active bindings that uses a substitute template (see the code) for length-one vectors and does not convert values to be array. 

Also the code on lines 92-99 in types_modelValues.R that initializes a modelValues object had the same issue, so I made a similar change there.

These changes pass test-models.R.  Let's run any other tests involving models before concluding there are no unintended consequences.
